sdk/src/types/block/output/rent.rs

```diff
@@ -1,224 +1,174 @@
+            .storage_score(*self);
+        let ed25519_address_score = null_address.storage_score(*self);
+        let basic_score_without_address = basic_output_score - ed25519_address_score;
+        let account_output_score = AccountOutputBuilder::new_with_amount(0, AccountId::null())
+            .add_unlock_condition(GovernorAddressUnlockCondition::new(null_address))
+            .add_unlock_condition(StateControllerAddressUnlockCondition::new(null_address))
+            .add_feature(
+                BlockIssuerFeature::new(0, vec![BlockIssuerKey::Ed25519(Ed25519BlockIssuerKey::null())]).unwrap(),
+            )
+            .storage_score(*self);
+        account_output_score - basic_score_without_address
     }
 }
 
-impl<T: Rent, const N: usize> Rent for [T; N] {
-    fn weighted_bytes(&self, config: RentStructure) -> u64 {
-        self.iter().map(|elem| elem.weighted_bytes(config)).sum()
-    }
-}
-
-pub struct MinimumStorageDepositBasicOutput {
-    config: RentStructure,
-    token_supply: u64,
-    builder: BasicOutputBuilder,
-}
-
-impl MinimumStorageDepositBasicOutput {
-    pub fn new(config: RentStructure, token_supply: u64) -> Self {
-        Self {
-            config,
-            token_supply,
-            builder: BasicOutputBuilder::new_with_amount(Output::AMOUNT_MIN).add_unlock_condition(
-                AddressUnlockCondition::new(Address::from(Ed25519Address::from([0; Ed25519Address::LENGTH]))),
-            ),
-        }
-    }
-
-    pub fn with_native_tokens(mut self, native_tokens: impl Into<Option<NativeTokens>>) -> Self {
-        if let Some(native_tokens) = native_tokens.into() {
-            self.builder = self.builder.with_native_tokens(native_tokens);
-        }
-        self
-    }
-
-    pub fn with_storage_deposit_return(mut self) -> Result<Self, Error> {
-        self.builder = self
-            .builder
-            .add_unlock_condition(StorageDepositReturnUnlockCondition::new(
-                Address::from(Ed25519Address::from([0; Ed25519Address::LENGTH])),
-                Output::AMOUNT_MIN,
-                self.token_supply,
-            )?);
-        Ok(self)
-    }
-
-    pub fn with_expiration(mut self) -> Result<Self, Error> {
-        self.builder = self.builder.add_unlock_condition(ExpirationUnlockCondition::new(
-            Address::from(Ed25519Address::from([0; Ed25519Address::LENGTH])),
-            1,
-        )?);
-        Ok(self)
+/// A trait to facilitate the computation of the byte cost of block outputs, which is central to dust protection.
+pub trait StorageScore {
+    fn storage_score(&self, _params: RentParameters) -> u64 {
+        0
     }
 
-    pub fn finish(self) -> Result<u64, Error> {
-        Ok(self.builder.finish_output(self.token_supply)?.rent_cost(self.config))
+    /// Computes the minimum deposit of this instance given [`RentParameters`].
+    fn min_deposit(&self, params: RentParameters) -> u64 {
```

Comment:
Thinking out loud, this is kinda only relevant to outputs, maybe it shouldn't be part of the trait

Reply:
I renamed the fn because it's really not the min deposit after all, it's just the rent cost. For instance, we use it to get the cost of just the SDRUC.